### PR TITLE
soc: intel_adsp: Generalize bootloader

### PR DESCRIPTION
**Pull request by finikorg**
Move adsp bootloader to soc/xtensa/intel_adsp making it available for other
boards.

### CODE REVIEW

File: soc/xtensa/intel_apl_adsp/CMakeLists.txt [hunk 1]
```diff
@@ -7,3 +7,5 @@ zephyr_library_sources(soc.c)
 zephyr_library_sources(main_entry.S)
 
 zephyr_library_sources_ifdef(CONFIG_SMP soc_mp.c)
+
+add_subdirectory(common)
```
Comment (nashif) [inline soc/xtensa/intel_apl_adsp/CMakeLists.txt:11]: what is that? I do not see you adding anything into it.

**Review by nashif — CHANGES_REQUESTED** [review]

File: soc/xtensa/intel_apl_adsp/CMakeLists.txt (hunk 1 above)
Comment (finikorg) [reply soc/xtensa/intel_apl_adsp/CMakeLists.txt:11]: Sorry, this was the result of cherry-pick from https://github.com/thesofproject/zephyr/commit/25812de5c1503031d3fb54e8e5fccb1ab3149041
Fixed now.

File: soc/xtensa/intel_apl_adsp/bootloader.cmake [hunk 2]
```diff
@@ -2,29 +2,27 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-if(EXISTS ${BOARD_DIR}/bootloader/CMakeLists.txt)
+if(EXISTS ${SOC_DIR}/${ARCH}/${SOC_FAMILY}/common/bootloader/CMakeLists.txt)
```
Comment (nashif) [inline soc/xtensa/intel_apl_adsp/bootloader.cmake:5]: doesn't this path always exist, in what cases we do not have this file?

Comment (finikorg) [reply soc/xtensa/intel_apl_adsp/bootloader.cmake]: I made it more simple.

**Review by andyross — COMMENTED** [review]
Cool with me, though the cherry-picking between this and the SOF tree is making my head spin. :)

**Review by nashif — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**